### PR TITLE
Bug 1956826: add restriction note for docker strategy build args

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 

/assign @rolfedh

### CODE REVIEW

**Comment by gabemontero** [comment]
@rolfedh - IMO we should also add an analogous note in https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies and all the same spots for 4.8, 4.6, 4.5, etc. where releases are still in support.

No need to update versions that are end of life.

Can you craft a PR for that ?

**Comment by gabemontero** [comment]
remembered we will also need an approver from https://github.com/openshift/api/blob/master/OWNERS as well ... will sort that out @adambkaplan @rolfedh once you all sign off on this change

File: build/v1/generated.proto
Comment (rolfedh) [inline build/v1/generated.proto:736]:
```suggestion
  // NOTE: Only the 'name' and 'value' fields are supported. Any settings on the 'valueFrom' field
```

File: build/v1/types.go
Comment (rolfedh) [inline build/v1/types.go:733]:
```suggestion
	// NOTE: Only the 'name' and 'value' fields are supported. Any settings on the 'valueFrom' field
```

File: build/v1/zz_generated.swagger_doc_generated.go
Comment (rolfedh) [inline build/v1/zz_generated.swagger_doc_generated.go:353]:
```suggestion
	"buildArgs":               "buildArgs contains build arguments that will be resolved in the Dockerfile. See https://docs.docker.com/engine/reference/builder/#/arg for more details. NOTE: Only the 'name' and 'value' fields are supported. Any settings on the 'valueFrom' field are ignored.",
```

**Review by rolfedh — APPROVED** [review]
Made some suggestions. Otherwise, /lgtm

**Comment by adambkaplan** [comment]
likewise, make the caps change and I'll add lgtm

**Comment by gabemontero** [comment]
I'll need to squash the commits and redo the `make update` @rolfedh @adambkaplan  ... I'll post an update here once that is ready

**Review by adambkaplan — APPROVED** [review]
/lgtm

File: build/v1/types.go
Comment (deads2k) [inline build/v1/types.go:733]: Has it ever been honored before?  This sort of a comment is an indication that the schema being embedded isn't actually the schema we wanted: it was just similar at some point in time.  

If it has never been honored, then creating a type with the schema you want seems more appropriate.

Comment (gabemontero) [reply build/v1/types.go:733]: It has never been honored.  And absolutely in hindsight choosing corev1.EnvVar was not the best choice here.

Admittedly we went with this band aid since we are coming up on code freeze.  I'll ultimately defer to @adambkaplan on this, but at least for me it was too late in the cycle to take on a schema change.

WDYT @adambkaplan ?

Comment (deads2k) [reply build/v1/types.go:733]: it seems that builds are actually serialized as an API into an env var (as I recall), which a build could then rely upon.  We have to keep this field now.

In the future, it's still ok to rely on an upstream schema in cases where you are simply passing it around blindly (deploymentconfig) for instance.  But in cases where you are interpreting the meaning of fields, the schema is very rarely actually the upstream one.

**Comment by gabemontero** [comment]
> /approve

thanks

fwiw, the fact it is not honored goes back as far as 3.6:  https://github.com/openshift/origin/blob/release-3.6/pkg/build/apis/build/types.go#L803

As far as anyone can recall the test platform team's attempt to use it is the first time this lack of support was reported.

**Comment by rolfedh** [comment]
> @rolfedh - IMO we should also add an analogous note in https://docs.openshift.com/container-platform/4.7/cicd/builds/build-strategies.html#builds-strategy-docker-build-arguments_build-strategies and all the same spots for 4.8, 4.6, 4.5, etc. where releases are still in support.
> 
> No need to update versions that are end of life.
> 
> Can you craft a PR for that ?

PTAL at `modules/builds-strategy-docker-build-arguments.adoc` in https://github.com/openshift/openshift-docs/pull/32982/files

**Pull request merged**
